rules/react.js

```diff
@@ -53,6 +53,13 @@ module.exports = {
     // https://github.com/yannickcr/eslint-plugin-react/blob/843d71a432baf0f01f598d7cf1eea75ad6896e4b/docs/rules/forbid-dom-props.md
     'react/forbid-dom-props': ['off', { forbid: [] }],
 
+    // Enforce a specific function type for function components.
+    // https://github.com/yannickcr/eslint-plugin-react/blob/v7.20.0/docs/rules/function-component-definition.md
+    'react/function-component-definition': ['error', {
+      namedComponents: 'function-declaration',
+      unnamedComponents: 'function-declaration',
+    }],
```

Comment:
I'm not sure about this. I prefer to write this way inside a function component, and this would probably prevent it 🤔 

```
var Component = (props) => {
  return <div />;
};
```

Also, `unnamedComponents` should be `function-expression` instead.

Reply:
This was mainly related to https://twitter.com/dan_abramov/status/1255229440860262400?s=09

Reply:
Hmmms, true 🤔  But means inside class, we would also have to not use arrows then. We can try it out and see how many changes etc.

Reply:
No it's just on functions, class arrow functions will be good